nengo_viz/static/viz_hotkeys.js

```diff
@@ -0,0 +1,35 @@
+VIZ.Hotkeys = function () { 
+    var self = this;
+
+    document.addEventListener('keypress', function(ev) {
+        // undo with ctrl-z
+        if (ev.ctrlKey == true && ev.keyCode == 90) {
+            VIZ.netgraph.notify({ undo: "1" });
+        }
+        // redo with shift-ctrl-z
+        if (ev.ctrlKey == true && ev.shiftKey == true && ev.keyCode == 90) {
+            VIZ.netgraph.notify({ undo: "0" });
+        }
+        // redo with ctrl-y
+        if (ev.ctrlKey == true && ev.keyCode == 89) {
+            VIZ.netgraph.notify({ undo: "0" });
+        }
+        // run model with spacebar
+        if (ev.keyCode == 32) {
+            sim.on_pause_click();
+        }
+        // bring up help menu with ? 
+        if (ev.shiftKey && ev.keyCode == 63) {
```

Comment:
Why is this checked for the `shift` key? Not all keyboard layouts (e.g., mine) require the `shift` key to be pressed to produce `?`.